specs/pvs/corla.pvs

```diff
@@ -1361,28 +1347,87 @@ END audits
+%%% The core RLA algorithm has a three basic parameters. Note that we
+%%% encode all percentages in the natural way as non-negative real
+%%% numbers (where $1$ means $100\%$).
 
+%%% First, one must choose a ``simultaneous risk limit'' ($\alpha$)
+%%% to be applied across all contents under audit. This notion is
+%%% refined in the context of Colorado the \texttt{colorado\_rlas}
+%%% theory that follows this one.  The risk limit must be dictated by
+%%% statute and rules. An typical risk limit is, e.g., $10\%$.
 % @trace RDPR-6-Jul-2017 Section 25.1.6
 % @description The largest statistical probability that an incorrect
 % reported tabulation outcome is not detected and corrected in a
 % risk-limiting audit.
+  risk_limit: TYPE = {a: nonneg_real | a <= 1.0} % α
+
+%%% The other two parameters impact the operational efficiency of the
+%%% audit, not its risk.
+
+%%% The ``error inflation factor'' (called $\gamma$, whose value is
+%%% always over $100\%$) must be chosen. The larger the error
+%%% inflation factor, the larger the initial sample needs to be, but
+%%% the less additional counting will be required if the audit of the
+%%% sample finds errors. The specific value chosen for CDOS is
+%%% specified in \texttt{colorado\_rlas}. A typical error inflation
+%%% factor is, e.g., $110\%$.
+  error_inflation_factor: TYPE = {g: nonneg_real | 1.0 <= g} % γ
+
+%%% Finally, an ``error tolerance'' (called %\lambda$, whose value is
+%%% always under $100\%$) must be chosen. The larger the error
+%%% tolerance is, the larger the initial sample size will have to be
+%%% to give high confidence that, even though the error rate in the
+%%% \emph{sample} is relatively large, the error rate for the
+%%% \emph{contests} as a whole will be small. A typical error
+%%% tolerance is, e.g., $50\%$.
+  error_tolerance: TYPE = {l: nonneg_real | l < 1.0 } % λ
+% @note kiniry We must use l since lambda is a reserved word
+% in PVS and using non-ASCII characters such as λ is not supported in
+% PVS.
+%%% The sample size multiplier (called $\rho$) depends upon $\alpha$,
+%%% $\gamma$, and $\lambda$ through the following formula.
+  sample_size_multiplier(a: risk_limit,
+                         g: error_inflation_factor,
+                         l: error_tolerance): real =
+    -log_lisp(a) / (1/(2*g) + l*log_lisp(1-1/(2*g)))
+%%% Recall that, however the values of $\alpha$, $\gamma$, and
+%%% $\lambda$ are set, $\rho$ is determined once and for all. It does
+%%% not depend upon the margins, the number or sizes of the
+%%% individuals contests, or on the audit data.
+
+%%% Next, we need the diluted margin, called $\mu$ in the
+%%% literature. The diluted margin is the smallest margin in votes
+%%% among the contests under audit (which we will call the
+%%% \texttt{minimum\_margin}) divided by the total number of ballots
+%%% cast across all the contests under audit.\todonealmcb{Is this
+%%% true, or is it the smallest margin over all contests under audit?}
```

Comment:
In my revisions that I'll be committing in a few minutes, I have refactored this definition in the `colorado_rlas` theory to the appropriate structures for Colorado's case.